source/std.tex

```diff
@@ -49,6 +49,15 @@
            ]{hyperref}
 \usepackage{memhfixc}     % fix interactions between hyperref and memoir
 
+
+%% Grammar extraction.
+\IfFileExists{./std-gram.ext}{
+\usepackage[extract-env={bnftab,simplebnf,bnf,bnfkeywordtab}]{extract}
+\def\PrintGrammar{\input{std-gram.ext}}}{
+\usepackage[active,header=false,handles=false,copydocumentclass=false,generate=std-gram.ext,extract-env={bnftab,simplebnf,bnf,bnfkeywordtab}]{extract}
+\def\PrintGrammar{Run again to include grammar summary}
```

Comment:
Changed, please take another look. I'm just calling into `extract` now to close the output file and reopen it under a new dummy name. The dummy file is discarded, but the point is that the main content is now always generated in the same run.